cmd/hosts.go

```diff
@@ -3,48 +3,364 @@ package cmd
+	Example: `
+privx-cli hosts show [access flags] --id <HOST-ID>,<HOST-ID>
+	`,
+	SilenceUsage: true,
+	RunE:         hostShow,
+}
+
+func hostShow(cmd *cobra.Command, args []string) error {
+	api := hoststore.New(curl())
+	hosts := []hoststore.Host{}
+
+	for _, id := range strings.Split(hostID, ",") {
+		host, err := api.Host(id)
+		if err != nil {
+			return err
+		}
+		hosts = append(hosts, *host)
+	}
+
+	return stdout(hosts)
+}
+
+//
+//
+var hostUpdateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update host",
+	Long:  `Update host`,
+	Example: `
+privx-cli hosts update [access flags] JSON-FILE --id <HOST-ID>
+	`,
+	Args:         cobra.ExactArgs(1),
+	SilenceUsage: true,
+	RunE:         hostUpdate,
+}
+
+func hostUpdate(cmd *cobra.Command, args []string) error {
+	var updateHost hoststore.Host
+	api := hoststore.New(curl())
+
+	err := decodeJSON(args[0], &updateHost)
+	if err != nil {
+		return err
+	}
+
+	err = api.UpdateHost(hostID, &updateHost)
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+//
+//
+var hostDeleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Delete host",
+	Long:  `Delete host. Host ID's are separated by commas when using multiple values, see example`,
+	Example: `
+privx-cli hosts delete [access flags] --id <HOST-ID>,<HOST-ID>
 	`,
 	SilenceUsage: true,
-	RunE:         hosts,
+	RunE:         hostDelete,
 }
 
-func hosts(cmd *cobra.Command, args []string) error {
+func hostDelete(cmd *cobra.Command, args []string) error {
+	api := hoststore.New(curl())
+
+	for _, id := range strings.Split(hostID, ",") {
+		err := api.DeleteHost(id)
+		if err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+//
+//
+var hostResolveCmd = &cobra.Command{
+	Use:   "resolve",
+	Short: "Resolve host",
+	Long:  `Resolve service and address to a single host`,
+	Example: `
+privx-cli hosts resolve [access flags] JSON-FILE
+	`,
+	Args:         cobra.ExactArgs(1),
+	SilenceUsage: true,
+	RunE:         hostResolve,
+}
+
+func hostResolve(cmd *cobra.Command, args []string) error {
+	var service hoststore.Service
+	api := hoststore.New(curl())
+
+	err := decodeJSON(args[0], &service)
+	if err != nil {
+		return err
+	}
+
+	host, err := api.ResolveHost(service)
+	if err != nil {
+		return err
+	}
+
+	return stdout(host)
+}
+
+//
+//
+var hostDeployableCmd = &cobra.Command{
+	Use:   "deployable",
+	Short: "Set a host to be depoyable or undeployable",
+	Long:  `Set a host to be depoyable or undeployable. Host ID's are separated by commas when using multiple values, see example`,
+	Example: `
+privx-cli hosts deployable [access flags] --id <HOST-ID>,<HOST-ID> --status=true
+	`,
+	SilenceUsage: true,
+	RunE:         hostDeployable,
+}
+
+func hostDeployable(cmd *cobra.Command, args []string) error {
+	api := hoststore.New(curl())
+
+	for _, id := range strings.Split(hostID, ",") {
+		err := api.UpdateDeployStatus(id, deployStatus)
+		if err != nil {
+			return err
+		}
```

Comment:
else: print successfully updated hostID.
Rational is as before: output of the command should be a newline-delimited list of successfully updated hostIDs. When error occurs it is easier to spot where the failure comes from. I think we should follow how the docker commands behave in this regard